File: pwa-blazor/pwa-blazor/Models/NodeInstance.cs
using System.Collections.Generic;
namespace pwa_blazor.Models
{
    public class NodeInstance
    {
        public int Data { get; set; }
        public List<NodeInstance> ChildNodes { get; set; }

        public NodeInstance(int data)
        {
            Data = data;
            ChildNodes = new List<NodeInstance>();
        }
        public void AddChild(int data)
        {
            ChildNodes.Add(new NodeInstance(data));
        }
        public void AddToData(int amount)
        {
            Data += amount;
        }
    }
}

File: pwa-react/src/models/NodeInstance.js
class NodeInstance {
    constructor(data, id) {
        this.data = Number(data)
        this.childNodes = Array(0)
        this.id = Number(id)
    }
    toObject() {
        return {
            data : this.data,
            childNodes : this.childNodes,
            id : this.id
        }
    }
    addChild(element) {
        this.childNodes.push(element)
    }
    addToData(addition) {
        this.data += Number(addition)
    }
}

export default NodeInstance